### ISSUE TITLE
Backup: Add support for backing up custom covers

### ISSUE THREAD

**Issue by FennekFox** [mihonapp/mihon#16]
### Describe your suggested feature

In order to be future-proof for more common mass migrations between apps and forks, I would very appreciate a feature which would allow the backup and restoration of custom cover arts. This is currently one of the only things which gets lost in migration, while being used to a moderate degree.

### Other details

Similar issues and suggested solutions were mentioned in [these issues](https://github.com/tachiyomiorg/tachiyomi/issues/10177).

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an official extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/extensions/issues/new/choose).
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by dan-malprod** [comment]
For me, the cover.jpg files were kept (at least for local manga) but they didnt render until I opened each library entry.

**Comment by peepincreepin** [comment]
Agree. I have had to reinstall & restore backup while debugging/troubleshooting many times, and it is a pain to reset custom covers each and every time. This feature would be massively appreciated. 🙌

**Comment by mizu-048** [comment]
+1

**Comment by mizu-048** [comment]
while looking through the data files in my android today, i found this folder 
```
Android/data/app.mihon/files/covers/custom/
```
which does actually contain my custom covers, i am not sure but copying this folder or the entire covers folder(if you don't want to waste data downloading them again), might solve the issue for custom covers being lost?

**Comment by peepincreepin** [comment]
> while looking through the data files in my android today, i found this folder
> 
> ```
> Android/data/app.mihon/files/covers/custom/
> ```
> 
> which does actually contain my custom covers, i am not sure but copying this folder or the entire covers folder(if you don't want to waste data downloading them again), might solve the issue for custom covers being lost?

i can't even access that folder without a computer; i don't think it would work for many users

![Screenshot_20240208_202036_My Files](https://github.com/mihonapp/mihon/assets/83149496/3d6abf98-578a-40c1-a1fd-938e0c5daaa9)

**Comment by mizu-048** [comment]
you can easily access that folder with the following methods:
- a file manager that uses **SAF(Storage Access Framework)** like ([Fossify file manager](https://github.com/FossifyOrg/File-Manager) (fork of SMT file manager)
-  androids own file manager **DocumentsUI(DUI)** which doesn't have a shortcut so you will have to use a third party app to launch it, like [Activity Launcher](https://github.com/butzist/ActivityLauncher) (launch "com.android.documentsui.files.FilesActivity" you can find it by searching any one the words in it) or you can use a launcher like [Material File Manager](https://github.com/zhanghai/MaterialFiles) which instead of using SAF just opens the folder in DUI

i hope that clears it?

**Comment by ClearSights** [comment]
> while looking through the data files in my android today, i found this folder
> 
> ```
> Android/data/app.mihon/files/covers/custom/
> ```
> 
> which does actually contain my custom covers, i am not sure but copying this folder or the entire covers folder(if you don't want to waste data downloading them again), might solve the issue for custom covers being lost?

I tried this approach. Some custom cover files are correctly recognized, but some are mapped to the wrong manga entry.

**Comment by mizu-048** [comment]
>> I tried this approach. Some custom cover files are correctly recognized, but some are mapped to the wrong manga entry. 

i guess then the only option is to custom set covers from the copied folder manually

**Comment by Boy656** [comment]
i was going to make a new ticket but i checked here first


![Image](https://github.com/user-attachments/assets/02f40f3f-19a3-49cb-bd91-f58ba8364ec3)

adding a button to save covers when you do a full backup will result in an improvement to user experience for those who are do format to their phones or switch from custom rom to another or simply delete a mihon and reinstall it this will help them to start over quickly 


this will be very important for those who are setting a special pictures to any manhwa or manga they watched and do not want to make it again after reinstall the application so making backup will make you easy start from where you stopped last time so saves time.
.
this is a feature i need so might be a helpful for others or no but i want to make a special pictures for some manhwa who have a lewd or nsfw pictures even the manhwa is completely sfw just the source put it on the manhwa i will select some sfw pictures with a lot of good sfw looking so i can proudly showing my friends my manhwa list without a fear that see any lewd image and i do a lot of factory rest so its hard to keep replace the images one after one after some time maybe the library will be big so its will waste a lot of time so i asked here a new features 
sorry the other details was just explain why i asked for this feature so its a little long.

**Comment by DATTABANDARU** [comment]
When reinstalling it is just hard to edit each and every cover manually for my entire library with over 200 different mangas,I doesn't like the low quality covers in my library preview which looks awful so I will put custom cover for  each and every entry by saving high quality covers with a lot of time but when the app needs to be reinstalled due crashing or something like that it is just not possible to do it again so it is really a important feature so please consider bringing it , it ok even if you set a thing like where all my custom covers get saved to a single folder.
Thankyou.

**Comment by shunsui18** [comment]
I also very much appreciate a dedicated custom cover backup feature.

**Comment by Smol-Ame** [comment]
"+1" and "I want this" is pointless & unnecessary, it also pings the Discord server & emails everyone subscribed to the request for no reason.

If someone wants to contribute the feature, then they can do so with their own time & efforts whenever they feel like it. Subscribe to this request, give it a thumbs up (or heart), and wait patiently & quietly if you don't have any useful information that will help contributors out.
